lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.PlantLoop.rb

```diff
@@ -91,32 +91,72 @@ def plant_loop_apply_prm_number_of_cooling_towers(plant_loop)
   end
 
   # Splits the single chiller used for the initial sizing run
-  # into multiple separate chillers based on Appendix G.
-  #
+  # into multiple separate chillers based on Appendix G. Also applies
+  # EMS to stage chillers properly
   # @param plant_loop [OpenStudio::Model::PlantLoop] chilled water loop
-  # @param sizing_run_dir [String] sizing run directory
   # @return [Boolean] returns true if successful, false if not
-  def plant_loop_apply_prm_number_of_chillers(plant_loop, sizing_run_dir = nil)
+  def plant_loop_apply_prm_number_of_chillers(plant_loop)
     # Skip non-cooling plants & secondary cooling loop
     return true unless plant_loop.sizingPlant.loopType == 'Cooling'
     # If the loop is cooling but it is a secondary loop, then skip.
     return true if plant_loop.additionalProperties.hasFeature('is_secondary_loop')
 
-    # Determine the number and type of chillers
-    num_chillers = nil
-    chiller_cooling_type = nil
-    chiller_compressor_type = nil
-
     # Set the equipment to stage sequentially or uniformload if there is secondary loop
     if plant_loop.additionalProperties.hasFeature('is_primary_loop')
       plant_loop.setLoadDistributionScheme('UniformLoad')
     else
       plant_loop.setLoadDistributionScheme('SequentialLoad')
     end
 
+    model = plant_loop.model
+
+    # Get all existing chillers and pumps. Copy chiller properties needed when duplicating existing settings
+    chillers = []
+    pumps = []
+    default_cop = nil
+    condenser_water_loop = nil
+    dsgn_sup_wtr_temp_c = nil
+
+    plant_loop.supplyComponents.each do |sc|
+      if sc.to_ChillerElectricEIR.is_initialized
+        chiller = sc.to_ChillerElectricEIR.get
+
+        # Copy the last chillers COP, leaving chilled water temperature, and reference cooling tower. These will be the
+        # default for any extra chillers.
+        default_cop = chiller.referenceCOP
+        dsgn_sup_wtr_temp_c = chiller.referenceLeavingChilledWaterTemperature
+        condenser_water_loop = chiller.condenserWaterLoop
+        chillers << chiller
+
+      elsif sc.to_PumpConstantSpeed.is_initialized
+        pumps << sc.to_PumpConstantSpeed.get
+      elsif sc.to_PumpVariableSpeed.is_initialized
+        pumps << sc.to_PumpVariableSpeed.get
+      end
+    end
+
+    # Get existing plant loop pump. We'll copy this pumps parameters before removing it. Throw exception for multiple pumps on supply side
+    if pumps.size.zero?
+      OpenStudio.logFree(OpenStudio::Error, 'prm.log', "For #{plant_loop.name}, found #{pumps.size} pumps. A loop must have at least one pump.")
+      return false
+    elsif pumps.size > 1
+      OpenStudio.logFree(OpenStudio::Error, 'prm.log', "For #{plant_loop.name}, found #{pumps.size} pumps, cannot split up per performance rating method baseline requirements.")
+      return false
+    else
+      original_pump = pumps[0]
+    end
+
+    return true if chillers.empty?
```

Comment:
why do we return `true` if an error? This seems to be part of the existing code but I am not sure why it is true.
Does the boolean has any meaning?

Reply:
I'm unsure-- I followed the convention from earlier in the function call to maintain consistency. It appears that this function returns `true` if any part of the plant loop suggests it does not apply:

https://github.com/NREL/openstudio-standards/compare/master...primary_secondary_loops#diff-5735b63aaf6176020260a80f43cc69fa1ffc4c9a4f5e44f4524fd70b69ee2100R99-R102